### PR TITLE
Config with pure python

### PR DESCRIPTION
**Pull request by icemac**
* Lint the code.
* Add support for Python 3.8 and 3.9.

### CODE REVIEW

File: src/zc/sourcefactory/browser/mapping.py [hunk 1]
```diff
@@ -24,12 +24,12 @@
...
 @zope.interface.implementer(zope.browser.interfaces.ITerms)
 class MappedTerms(object):
-    """A terms implementation that knows how to handle a source that was 
+    """A terms implementation that knows how to handle a source that was
     created through a source factory.
     """
 
     zope.component.adapts(zc.sourcefactory.mapping.ValueMappingSource,
-        zope.publisher.interfaces.browser.IBrowserRequest)
+                          zope.publisher.interfaces.browser.IBrowserRequest)
```
Comment (sallner) [inline src/zc/sourcefactory/browser/mapping.py:32]: That suprises me actually, I thought `zope.component.adapts` was not possible with Python 3. I guess we have untested code here. There should not be the claim of Python 3 support unless this package is properly ported.

File: src/zc/sourcefactory/policies.py [hunk 2]
```diff
@@ -54,7 +55,8 @@ def getTitle(self, value):
         if md:
             title = md.title
         else:
-            title = value.decode() if isinstance(value, bytes) else unicode(value)
+            title = value.decode() if isinstance(value,
+                                                 bytes) else unicode(value)
```
Comment (sallner) [inline src/zc/sourcefactory/policies.py:59]: Interesting way to format. Maybe it would be more clear with a proper if/else structure.

**Review by sallner — CHANGES_REQUESTED** [review]
I would prefere, if all `adapts` and `implements` are removed from the code base of this package. Maybe we can also delete the `buildout.cfg` here as the `[py]` section does not seem to important to me.

File: src/zc/sourcefactory/browser/mapping.py (hunk 1 above)
Comment (icemac) [reply src/zc/sourcefactory/browser/mapping.py:32]: According to https://github.com/zopefoundation/zope.component/blob/e7fe69a371afd52781a3c49c130768b607bcff48/src/zope/component/_declaration.py#L39-L50 it should still work.

And yes, `src/zc/sourcefactory/browser/mapping.py` is mostly untested but this statement is executed at import time and does not raise an error.

Comment (icemac) [reply src/zc/sourcefactory/browser/mapping.py:32]: But it should be refactored, you are right.

File: src/zc/sourcefactory/policies.py (hunk 2 above)
Comment (icemac) [reply src/zc/sourcefactory/policies.py]: Fixed in https://github.com/zopefoundation/zc.sourcefactory/pull/7/commits/9928a0a306dc11092171aa039c18b8048f3d63d9

**Review by sallner — APPROVED** [review]
I approve these changes.

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**